src/test/java/com/uid2/admin/store/writer/EncryptedScopedStoreWriterTest.java

```diff
@@ -0,0 +1,385 @@
+                byte[] encryptedPayload = AesGcm.encrypt(data.getBytes(StandardCharsets.UTF_8), secret);
+                JsonObject encryptedJson = new JsonObject()
+                        .put("key_id", largestKey.getId())
+                        .put("encryption_version", "1.0")
+                        .put("encrypted_payload", Base64.getEncoder().encodeToString(encryptedPayload));
+
+                // Define the location for the encrypted file
+                CloudPath encryptedFilePath = new CloudPath(sitesDir).resolve("sites_encrypted.json");
+
+                // Call fileManager to upload the encrypted data
+                fileManager.uploadFile(encryptedFilePath, dataFile, encryptedJson.encodePrettily());
+
+                // Verify the result
+                assertTrue(cloudStorage.list(sitesDir).contains(sitesDir + "/sites_encrypted.json"));
+            } else {
+                throw new IllegalStateException("No S3 keys available for encryption for site ID: " + testSiteId);
+            }
+        }
+    }
+
+
+
+    // Helper method to check if the data is already encrypted
+    private boolean isEncrypted(String data) {
+        try {
+            JsonObject json = new JsonObject(data);
+            return json.containsKey("key_id") &&
+                    json.containsKey("encryption_version") &&
+                    json.containsKey("encrypted_payload");
+        } catch (Exception e) {
+            return false;
+        }
+    }
+
+
+    @Test
+    void testUploadWritesEncryptedFileToCorrectLocation() throws Exception {
+        String data = "test data";
+        JsonObject extraMeta = new JsonObject();
+
+        // Directly execute the logic from uploadWithEncryptionKey
+        byte[] secret = Base64.getDecoder().decode(encryptionKey.getSecret());
+        byte[] encryptedPayload = AesGcm.encrypt(data.getBytes(StandardCharsets.UTF_8), secret);
+        JsonObject encryptedJson = new JsonObject()
+                .put("key_id", encryptionKey.getId())
+                .put("encryption_version", "1.0")
+                .put("encrypted_payload", Base64.getEncoder().encodeToString(encryptedPayload));
+
+        // Define the location for the encrypted file
+        CloudPath encryptedFilePath = new CloudPath(sitesDir).resolve("sites_encrypted.json");
+
+        // Ensure the file does not exist before upload (clean up any existing state)
+        List<String> files = cloudStorage.list(sitesDir);
+        String encryptedFilePathStr = encryptedFilePath.toString();
+        if (files.contains(encryptedFilePathStr)) {
+            cloudStorage.delete(encryptedFilePathStr);
+        }
+
+        // Call fileManager to upload the encrypted data
+        fileManager.uploadFile(encryptedFilePath, dataFile, encryptedJson.encodePrettily());
+
+        // Verify the result
+        assertThat(cloudStorage.list(sitesDir)).contains(sitesDir + "/sites_encrypted.json");
```

Comment:
This is only one of the asserts I would expect - either in this test or another one. I would expect to see that the file that has been written contains the correct data, and that the data can be decrypted correctly.